pandas/tests/indexing/test_coercion.py

```diff
@@ -580,12 +580,11 @@ def test_where_series_datetime64(self, fill_val, exp_dtype):
         values = pd.Series(pd.date_range(fill_val, periods=4))
         if fill_val.tz:
             exp = pd.Series([pd.Timestamp('2011-01-01'),
-                             pd.Timestamp('2012-01-02 05:00'),
```

Comment:
what changed here?

Reply:
Previous behavior was was treating `other` for DatetimeTZ as ints (see the writeup for #21546). As a result was getting incorrectly merged with naive datetime Series as naive UTC timestamps. Fixing the behavior of tz-aware `other` also fixed this and so now the result of `combine_first` is a promotion of the whole Series to `object` (as was desired per the deleted xfail/TODO).